test/e2e/customconfigs/waitforpodsready_test.go

```diff
@@ -99,284 +126,360 @@ var _ = ginkgo.Describe("WaitForPodsReady Job Controller E2E", ginkgo.Ordered, f
 	})
 
 	ginkgo.AfterAll(func() {
+		util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, curlPod, true, util.LongTimeout)
 		util.ExpectObjectToBeDeleted(ctx, k8sClient, metricsReaderClusterRoleBinding, true)
 	})
 
-	ginkgo.When("WaitForPodsReady has a tiny Timeout and no RecoveryTimeout", func() {
-		ginkgo.BeforeEach(func() {
-			updateKueueConfiguration(func(cfg *configapi.Configuration) {
-				cfg.WaitForPodsReady = &configapi.WaitForPodsReady{
-					Enable:          true,
-					BlockAdmission:  ptr.To(true),
-					Timeout:         &metav1.Duration{Duration: util.TinyTimeout},
-					RecoveryTimeout: nil,
-					RequeuingStrategy: &configapi.RequeuingStrategy{
-						Timestamp:          ptr.To(configapi.EvictionTimestamp),
-						BackoffBaseSeconds: ptr.To(int32(10)),
-						BackoffLimitCount:  ptr.To(int32(1)),
-					},
-				}
-			})
-
-			curlPod = testingjobspod.MakePod("curl-metrics", configapi.DefaultNamespace).
-				ServiceAccountName(serviceAccountName).
-				Image(util.E2eTestAgnHostImage, util.BehaviorWaitForDeletion).
-				TerminationGracePeriod(1).
+	ginkgo.It("should evict and requeue workload when pods readiness timeout is surpassed", func() {
+		ginkgo.By("creating a suspended job so its pods never report Ready", func() {
+			job = testingjob.MakeJob("job-timeout", ns.Name).
+				Queue(kueue.LocalQueueName(lq.Name)).
+				Request(corev1.ResourceCPU, "2").
+				Parallelism(1).
 				Obj()
-			util.MustCreate(ctx, k8sClient, curlPod)
+			gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
```

Comment:
```suggestion
			util.MustCreate(ctx, k8sClient, job)
```

Shouldn't we use the `util.MustCreate`?

Reply:
Yep, we can. Let me update it everywhere.

Reply:
Thank you.
/lgtm
/hold cancel

Reply:
Done